File: LeetCode/python3/1029.py
class Solution:
    def twoCitySchedCost(self, costs: List[List[int]]) -> int:
        costs.sort(key=lambda x:x[0] - x[1])
        sum, n = 0, len(costs) // 2
        for i in range(n * 2):
            sum += costs[i][0] if i < n else costs[i][1]
        return sum
        

File: LeetCode/cpp/1029.cpp
class Solution {
public:
    int twoCitySchedCost(vector<vector<int>> &costs) {
        int sum = 0, n = costs.size();
        sort(costs.begin(), costs.end(), [](const vector<int> &c1, const vector<int> &c2) {
            return c1[0] - c1[1] < c2[0] - c2[1];
        });
        for (int i = 0; i < n; ++i)
            sum += (i < n / 2 ? costs[i][0] : costs[i][1]);
        return sum;
    }
};
